modules/standard/IO.chpl

```diff
@@ -9511,10 +9511,11 @@ proc fileReader.readBinary(ref b: bytes, maxSize: int): bool throws {
   * when ``true``: the new methods are called. These return an ``int`` with the
     number of values that were read.
 
-  This ``param`` can be set to true by compiling your program with
-  ``-sReadBinaryArrayReturnInt=true``
+  .. warning::
+
+    This flag is deprecated and will be removed in a future release.
 */
-config param ReadBinaryArrayReturnInt = false;
+config param ReadBinaryArrayReturnInt = true;
```

Comment:
I'd expect you'd use `@deprecated` here. I think, if somebody updated their Makefile to respond to the last release, and they have `-sReadBinaryArrayReturnInt=true`, then we want to issue a deprecation warning that indicates the default is now `true` and that the flag will be removed in the future. Is that easy enough to do? Am I missing something?

Reply:
I went with that strategy initially; however, since it's a top-level symbol, a deprecation message gets emitted whenever IO is imported. So I decided to just indicate that it's deprecated in the deprecation messages for `readBinary` and in the docs.

Reply:
That seems very odd to me. We just want to get a deprecation when using it from the command line & not from the uses of it in the standard library. Is there something wrong with the compiler's deprecation handling?

We could put the deprecation warning in the compiler's handling of config params, if we had no better way.